extra/onnx_ops.py

```diff
@@ -96,14 +96,16 @@ def Conv(X, W, B=None, auto_pad="NOTSET", dilations=1, group=1, kernel_shape=Non
 def ConvTranspose(X, W, B=None, auto_pad="NOTSET", dilations=1, group=1, kernel_shape=None, pads=None, strides=1):
   return X.conv_transpose2d(W, B, stride=strides, groups=group, dilation=dilations, padding=(pads[1], pads[3], pads[0], pads[2]) if pads is not None else 0)
 
-# TODO: copied from tensor.py
 def Dropout(data, ratio=0.5, training_mode=False, seed=None):
-  # TODO: mask should be a boolean tensor
-  if not training_mode: return data, Tensor.ones(*data.shape)  # if mask is requested as output it will contain all ones.
-  if seed is not None: Tensor.manual_seed(seed)
-  _mask : np.ndarray = np.asarray(Tensor._rng.binomial(1, 1.0-ratio, size=data.shape), dtype=data.dtype)
-  mask = Tensor(_mask, requires_grad=False, device=data.device)
-  return data * mask * (1/(1.0 - ratio)), mask
+  class _FakeRNG:
+    def __init__(self, seed): self.seed = seed
+    def random(self, size=None, dtype=None):
+      ret = np.random.RandomState(self.seed).random(size)
+      return ret.astype(dtype) if dtype else ret
+  Tensor.training = training_mode
+  if seed is not None: Tensor._rng = _FakeRNG(seed)  # Need to use RandomState because of how test data gets generated.
```

Comment:
We need something better than this. This affects all future rng in a weird way.

Reply:
True that! I hadn't considered it would stick to `Tensor`. Will fix in a bit.

Reply:
Added a param to `Tensor.manual_seed` to support [legacy random generation](https://numpy.org/doc/stable/reference/random/legacy.html).